### ISSUE TITLE
remove Activestate login banner

### ISSUE THREAD

**Issue by perfecto25** [ActiveState/OpenKomodoIDE#4]
- wanted to try this IDE but it requires registration w AS, will submit PR to remove login banner

**Comment by th3coop** [comment]
I suggest not removing it but allowing one to bypass it.  The login allows for pretty handy integration with the ActiveState's Platform, eg. automatic development env creation and activation when you open a project.

Let me know if you want me to point you where to look for the code.

**Comment by Sainteyegor** [comment]
Has there been any action taken on this issue? I'd like to be able to run Komodo-IDE on isolated networks.

Can you point me to where the relevant code lives?

**Comment by Naatan** [comment]
https://github.com/ActiveState/OpenKomodoIDE/blob/master/src/chrome/komodo/content/komodo.p.js#L456

Just follow the breadcrumbs from there. I think just dropping that line will get you what you want.

**Comment by agroszer** [comment]
Sitting here behind a corporate firewall/proxy. That sign in dialog tells me: "Login failed, it appears you are offline" What could be wrong? All other apps/firefox/chrome can access the internet.
Credentials are fine, I can login at https://platform.activestate.com/

**Comment by th3coop** [comment]
If you don't want to mess around rebuilding Komodo you can follow these instructions: https://community.komodoide.com/t/login-failed-it-appears-you-are-offline-again/5903/14

**Comment by agroszer** [comment]
yeah thanks, recompile is beyond me

**Comment by eborgbjerg** [comment]
I get the same error on an Ubuntu box.
The link about 7-zip on windows is obviously not applicable here.
So I guess my only option for using this IDE would be to change the source code and build it myself?

**Comment by th3coop** [comment]
@eborgbjerg, you're not out of luck!  Odd that you're the first Linux user to hit this though.  You should be able to perform the 7Zip step but using whatever zip tool Linux has.  I don't have a Linux machine around to workout the exact steps so I'm going to give broad instructions and info here.

The komodo.tar file is just a *.zip file with a different file extension. So in Linux, you'll do the same 7Zip step but use whatever (un)zip tool you have on Linux.

Key point to be aware of:
- Note that when you decompress the *komodo.jar* file, the **content**, **skin** and **locale** folders are in the CWD, NOT a folder called **komodo**.  When you've edited the **komodo.js** file and are ready to make the new **komodo.jar**, you need to make sure you create it so it decompresses the same way.  It's straight forward though.  Looks like it's just `zip -r komodo.jar content locale skin`.

If this doesn't work though, don't hesitate to ping back here and I'll try to get you sorted.

**Comment by eborgbjerg** [comment]
@th3coop thanks for your reply and explanation!

I did all 12 steps as accurately as I could, also tried creating the komodo.jar file both by linux File Manager and with the command you suggested.
But I still get the login window.

Looking at the code down in `chrome/content/login/login.js`
it seems this is where the text
"Sign in to your ActiveState Platform account" comes from.
Is it possible that this function could be activated in a way that got added to the code base after your original answer from 2022?

**Comment by th3coop** [comment]
> Is it possible that this function could be activated in a way that got added to the code base after your original answer from 2022?


It's possible and was in fact true, until about 2 weeks ago.  I updated the instructions 2 weeks ago after someone asked a similar question.

Another Linux user commented on the thread recently that they had to rename their *~/.komodoide* folder to get the change to be picked up.

So two things to try here:
1. Have a look at the instructions again and make sure you followed the updated one.  The commented line is now line 27 (less scrolling than line 400 something).
2. Try what gazer1912 suggested in the thread but note my comment below theirs about keeping your profile folder contents.
    1. Gazer's comment: https://community.komodoide.com/t/login-failed-it-appears-you-are-offline-again/5903/28

**Comment by eborgbjerg** [comment]
Yep, gazer1912's suggestion solved it.
Probably something gets cached in the ~/.komodoide directory.

**Comment by Typel** [comment]
Because links don't last forever and all of the above forum links seem to have died, I dredged them up from old archives and summarized below for posterity. These worked for me as of Feb 2025.

1. Install Komodo IDE 12 to your preferred location.
2. Open up that location, for example /home/your-name/Komodo-IDE
3. Navigate to the subdirectory /lib/mozilla/chrome within your install dir.
4. Extract the contents of komodo.jar - it will contain 3 folders (content, locale, skin)
5. From the extracted folders, open up content/komodo.js in a text editor.
6. Comment out line 27 which should read `auth.authenticated((authenticated) => if ( ! authenticated) openLoginDialog();});` and save.
7. Compress those 3 folders back up into a zip file, rename it to komodo.jar
8. Replace the original komodo.jar (you might want to back up the original first)
9. Restart your computer (I found this to be necessary before it took effect)
10. Start up Komodo IDE - there should no longer be any login prompt.

**Comment by keravesque** [comment]
Thank you for documenting those steps @Typel - I just followed them but was still seeing the prompt and still unable to close it without the IDE closing on Linux Mint 22.1.

I noticed mention of a ~./komodoide folder, but unfortunately the link to that comment is also now broken and WayBack Machine doesn't have that URL cached. I did a little digging myself and found several cache directories, and took a blind guess at what to try deleting, and luckily my intuition seems to have served me well, as now the program is starting without the login prompt!

For anyone else who comes here with this issue, I delete the contents of:
~/.komodoide/12.0/XRE/cache2/entries
~/.komodoide/12.0/XRE/startupCache

I don't know which files were actually responsible, but the deleted files included startupCache.8.little and three extensionless files with names that look like SHA-1 hashes without hyphens, or something of the sort.

**Comment by keravesque** [comment]
For posterity I figure it might make sense to have one single comment with all the instructions to make this work. These steps worked for me on Linux Mint 22.1 as of the time of this writing:

1. Install Komodo IDE 12 to your preferred location.
2. Open up that location, for example /home/your-name/Komodo-IDE
3. Navigate to the subdirectory /lib/mozilla/chrome within your install dir.
4. Extract the contents of komodo.jar - it will contain 3 folders (content, locale, skin)
5. From the extracted folders, open up content/komodo.js in a text editor.
6. Comment out line 27 which should read auth.authenticated((authenticated) => if ( ! authenticated) openLoginDialog();}); and save.
7. Compress those 3 folders back up into a zip file, rename it to komodo.jar
8. Replace the original komodo.jar (you might want to back up the original first)
9. Restart your computer (not sure if needed, but I did it)
10. Delete the contents of the following directories:
- [ ] ~/.komodoide/12.0/XRE/cache2/entries
- [ ] ~/.komodoide/12.0/XRE/startupCache
12. Start up Komodo IDE - there should no longer be any login prompt.

If you are using Windows, apparently the directory where Komodo keeps its cache is usually:
C:\Users\[YOU]\AppData\Local\ActiveState\KomodoIDE[version] 

I hope this helps someone!

(Edited to include the correct Windows path provided by @th3coop )

**Comment by th3coop** [comment]
@keravesque I really appreciate you posting and your patience.

For Windows the path is usually **C:\Users\[YOU]\AppData\Local\ActiveState\KomodoIDE\[version]\**.

The `KOMODO_USERDATADIR` isn't set unless you manually set it.  It's used to override the default profile location, usually for testing, or for installing multiple instances of Komodo.

**Comment by keravesque** [comment]
Happy to contribute whatever I can - it seems pretty great so far! I've been looking for a good PHP IDE for ages, and although I just recently started using WebStorm for JS after hearing lots of good things about it, there are some things I find pretty irritating about WS, so using Komodo as a Javascript and Typescript IDE is an appealing prospect to me as well.

Vue and React support out of the box would be nice, but given the open source state of things, that seems entirely possible to make happen. I was glad to see this blog post still available on Wayback Machine that should help me get React setup and which I can probably leverage to do something very similar for Vue: https://web.archive.org/web/20240420185908/https://www.activestate.com/blog/jsx-and-react-komodo-syntax-highlighting-case-study/

**Comment by th3coop** [comment]
Rad!  Very cool to hear your motivation.  Since we don't have the forums anymore I'd suggest creating new Issues in this repo even if your just asking clarifying questions.  The former lead dev and myself still watch the repo to help out where we can.

Be warned though that Komodo is now running on a VERY outdated version of Mozilla and will be hard if not impossible to actually build if that's required for any work you do.  I don't want you to invest a ton of time in Komodo just to hit a hard stop down the road.

Having said that, most any customizations you do to Komodo do NOT require a rebuild and can be done through the Addons mechanic.  The entire program could be re-implemented entirely through addons.

**Comment by ducna-vbee** [comment]
I've managed to login to ActiveState in Komodo IDE in my Windows computers. It requires adding exceptions to server certificates in Preferences.

**Comment by rnelson** [comment]
> I've managed to login to ActiveState in Komodo IDE in my Windows computers. It requires adding exceptions to server certificates in Preferences.

Preferences > Web & Browser > Connection > View Certificates > Servers > Add Exception...

Enter "https://platform.activestate.com" and click Get Certificate, ensure that "Permanently store this exception" is checked, and click Confirm Security Exception. OK, Apply/OK, and now logins should work.

**Comment by ducna-vbee** [comment]
> > I've managed to login to ActiveState in Komodo IDE in my Windows computers. It requires adding exceptions to server certificates in Preferences.
> 
> Preferences > Web & Browser > Connection > View Certificates > Servers > Add Exception...
> 
> Enter "https://platform.activestate.com" and click Get Certificate, ensure that "Permanently store this exception" is checked, and click Confirm Security Exception. OK, Apply/OK, and now logins should work.

Yes. That's the way to sneak into the sign-in page. Thanks for the explicit clarification!
